### PR TITLE
Fix the glyph writer

### PR DESCRIPTION
**Pull request by carado**
Hi;
I was getting some truncated zeroes out of the glyph writer.
On my end, this code produces correct results and passes tests.

### CODE REVIEW

File: src/writer/writer.rs [hunk 1]
```diff
@@ -104,22 +104,18 @@ impl<T: Write> Writer<T> {
...
-						value  |= if map.get(x, y) { 1 } else { 0 };
-					}
-
-					value <<= 8 - (map.width() % 8);
-
-					let hex = format!("{:X}\n", value);
-
-					if (hex.len() - 1) % 2 != 0 {
-						write!(self.stream, "0");
+					for x_char in 0 .. (map.width() + 3) / 4 {
```
Comment (meh) [inline src/writer/writer.rs:107]: Where does the `3` come from? Are you sure this is not fixing it for the font you're using and breaking it for every other font?

Feels like there's definitely an issue tho.

**Review by meh — CHANGES_REQUESTED** [review]

File: src/writer/writer.rs (hunk 1 above)
Comment (carado) [reply src/writer/writer.rs:107]: `(x + 3) / 4` is how I do the integer equivalent to `ceil(x / 4)`; one needs a number of hex characters (4 bits each) that is the number of bits divided by four, rounded up.

Comment (meh) [reply src/writer/writer.rs:107]: Could you add a comment clarifying that please? Or do the integer/float/integer dance, I'd rather have more clarity than more performance in this case :panda_face:

**Comment by meh** [comment]
Actually, I think https://github.com/meh/rust-bdf/pull/2/commits/480c0893d076b57fb94885c29a1eb24624567e14 is a better fix for this?

What do you think?

**Comment by carado** [comment]
That seems to work fine for my code as well; feel free to use it rather than my own fix.

**Pull request closed without merging**